typescript/core/src/ChunkCursor.ts

```diff
@@ -38,10 +38,6 @@ export class ChunkCursor {
     this.#startTime = params.startTime;
     this.#endTime = params.endTime;
     this.#reverse = params.reverse;
-
-    if (this.chunkIndex.messageIndexLength === 0n) {
```

Comment:
https://mcap.dev/spec#chunk-index-op0x08

I think it would be helpful/pedantic to confirm that the message start/end time are 0 and messageIndexOffsets is also empty? these should all be true if there are indeed no messages. Otherwise we should throw since _that_ would mean there are messages but no message index (or an incorrect length for the index) which indicates malformed file from the POV of our app or spec.

Reply:
Added an extra check in `loadMessageIndexes()`. I think this is a better place to throw rather than in the constructur

Reply:
Why not throw as early as possible?

Reply:
What makes you say it is a better place to throw?